packages/flutter/lib/src/cupertino/refresh.dart

```diff
@@ -0,0 +1,520 @@
+  /// where the sliver will start retracting.
+  final RefreshCallback onRefresh;
+
+  static const double _kDefaultRefreshTriggerPullDistance = 100.0;
+  static const double _kDefaultRefreshIndicatorExtent = 60.0;
+
+  /// Retrieve the current state of the CupertinoRefreshControl. The same as the
+  /// state that gets passed into the [builder] function. Used for testing.
+  @visibleForTesting
+  static RefreshIndicatorMode state(BuildContext context) {
+    final _CupertinoRefreshControlState state
+        = context.ancestorStateOfType(const TypeMatcher<_CupertinoRefreshControlState>());
+    return state.refreshState;
+  }
+
+  /// Builds a simple refresh indicator that fades in a bottom aligned down
+  /// arrow before the refresh is triggered, a [CupertinoActivityIndicator]
+  /// during the refresh and fades the [CupertinoActivityIndicator] away when
+  /// the refresh is done.
+  static Widget buildSimpleRefreshIndicator(BuildContext context,
+    RefreshIndicatorMode refreshState,
+    double pulledExtent,
+    double refreshTriggerPullDistance,
+    double refreshIndicatorExtent,
+  ) {
+    const Curve opacityCurve = const Interval(0.4, 0.8, curve: Curves.easeInOut);
+    return new Align(
+      alignment: Alignment.bottomCenter,
+      child: new Padding(
+        padding: const EdgeInsets.only(bottom: 16.0),
+        child: refreshState == RefreshIndicatorMode.drag
+            ? new Opacity(
+                opacity: opacityCurve.transform(
+                  min(pulledExtent / refreshTriggerPullDistance, 1.0)
+                ),
+                child: const Icon(
+                  CupertinoIcons.down_arrow,
+                  color: CupertinoColors.inactiveGray,
+                  size: 36.0,
+                ),
+              )
+            : new Opacity(
+                opacity: opacityCurve.transform(
+                  min(pulledExtent / refreshIndicatorExtent, 1.0)
+                ),
+                child: const CupertinoActivityIndicator(radius: 14.0),
+              ),
+      ),
+    );
+  }
+
+  @override
+  _CupertinoRefreshControlState createState() {
+    return new _CupertinoRefreshControlState();
+  }
+}
+
+class _CupertinoRefreshControlState extends State<CupertinoRefreshControl> {
+  /// Reset the state from done to inactive when only this fraction of the
+  /// original `refreshTriggerPullDistance` is left.
+  static const double _kInactiveResetOverscrollFraction = 0.1;
+
+  RefreshIndicatorMode refreshState;
+  // [Future] returned by the widget's `onRefresh`.
+  Future<void> refreshTask;
+  // The amount of space available from the inner indicator box's perspective.
+  //
+  // The value is the sum of the sliver's layout extent and the overscroll
+  // (which partially gets transfered into the layout extent when the refresh
+  // triggers).
+  //
+  // The value is independent from the slivers scrollOffset. i.e. it's still
```

Comment:
scrollOffset. i.e. it's --> scrollOffset: it's

Reply:
I meant more 'in other words'. ':' sounds like the rest is referring to scrollOffset rather than lastIndicatorExtent

Reply:
Then how about:
The value of `lastIndicatorExtent` doesn't change when the sliver scrolls away without retracting; it is independent from the sliver's `scrollOffset`.

When I read yours, it was unclear what "it's" was referring to.

Reply:
sg